### ISSUE TITLE
transaction==2.4.0 seems incompatible with WSGI-Publisher

### ISSUE THREAD

**Issue by dwt** [zopefoundation/Zope#390]
I'm not completely sure I've got it right, but it seems that this pull request https://github.com/zopefoundation/transaction/commit/d2b90a701a7d718e0a5f86dd85e50c818611c500 has broken this check https://github.com/zopefoundation/Zope/blob/master/src/ZPublisher/WSGIPublisher.py#L226

As far as I understand it, the Transaction manager is now a ThreadLocalWrapper and doesn't have this private attribute anymore.

**Comment by dwt** [comment]
The problem I'm seeing is that after startup, with the newest transaction manager Zope immediately explodes on the first request in the linked line.

**Comment by icemac** [comment]
@dwt Does it also fail with the changes made in #388?

**Comment by dataflake** [comment]
@dwt I cannot reproduce the issue on a recent checkout, and stepping through the code does not show any problems.

When I use `pdb` to look at the variables in the referenced code (see below) I see that `transaction.manager` is a `ThreadTransactionManager` that won't have the `_txn` attribute, but the code is looking at its `manager` attribute, which is a `TransactionManager` and has `_txn`.

https://github.com/zopefoundation/Zope/blob/4dc170079185e9cbc889efc691e6b40fe1533b59/src/ZPublisher/WSGIPublisher.py#L226

``` python
-> if transaction.manager.manager._txn is not None:
(Pdb) p transaction.manager
<transaction._manager.ThreadTransactionManager object at 0x10302e228>
(Pdb) p transaction.manager.manager
<transaction._manager.TransactionManager object at 0x1051f9ef0>
(Pdb) p transaction.manager.manager._txn
None
```

**Comment by icemac** [comment]
I believe this issue was fixed in 950a7855119966d8b52fd04ef9fd4b07f2772c36. `setup.py` on `master` requires `transaction >= 2.4`.
